src/components/Patient/Utils.ts

```diff
@@ -17,3 +25,38 @@ export function isPatientMandatoryDataFilled(patient: PatientModel) {
     patient.blood_group
   );
 }
+
+export const useAddPatientNote = (options: {
+  patientId: string;
+  consultationId?: string;
+}) => {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: async ({
+      noteField,
+      thread,
+      replyTo,
+    }: {
+      noteField: string;
+      thread: PatientNotesModel["thread"];
+      replyTo?: { id: string };
+    }) => {
+      const performMutation = mutate(routes.addPatientNote, {
+        pathParams: { patientId: options.patientId },
+      });
+
+      return performMutation({
+        note: noteField,
+        thread,
+        consultation: options.consultationId,
+        reply_to: replyTo?.id,
+      });
+    },
```

Comment:
why complicate things?

```suggestion
    mutationFn: mutate(routes.addPatinetNote, {
      pathParams: { patientId: options.patientId }
    }),
```